pkg/utils/grub.go

```diff
@@ -169,37 +170,89 @@ func (g Grub) Install(target, rootDir, bootDir, grubConf, tty string, efi bool,
 		}
 
 		// Copy needed files for efi boot
-		shimFiles := cnst.GetGrubFilePaths(g.config.Arch)
-
+		// This seems like a chore while we could provide a package for those bundled files as they are just a shim and a grub efi
+		// BUT this is needed for secureboot
+		// The shim contains the signature from microsoft and the shim provider (i.e. upstream distro like fedora, suse, etc)
+		// So if we use the shim+grub from a generic package (i.e. ubuntu) it WILL boot with secureboot
+		// but when loading the kernel it will fail because the kernel is not signed by the shim provider or grub provider
+		// the kernel signature would be from fedora while the shim signature would be from ubuntu
+		// This is why if we want to support secureboot we need to copy the shim+grub from the rootfs default paths instead of
+		// providing a generic package
+		shimFiles := utils.GetEfiShimFiles(g.config.Arch)
```

Comment:
I see some similitudes here with https://github.com/kairos-io/enki/pull/17/files

Like on the other PR, I would recommend to have different functions for the ship and grub processes. This way maybe we can see some commonalities that can be abstracted later on, but even if that's not the case I think it makes the code easier to read and review

Reply:
yes, I wanted to test it in both places and maybe search for a way to centralize it on a helper function in the sdk so it doesnt diverge too much.

I will take a look and see if its possible, otherwise I will extract those things into a couple of functions